e2e/src/app.po.ts

```diff
@@ -2,10 +2,72 @@ import { browser, by, element } from 'protractor';
 
 export class AppPage {
   navigateTo() {
-    return browser.get(browser.baseUrl) as Promise<any>;
+    return browser.get('/');
   }
 
-  getTitleText() {
-    return element(by.css('app-root .content span')).getText() as Promise<string>;
+  getActionItemsTable() {
+    return element(by.css('.table-container'));
+  }
+
+  getActionItems() {
+    return element.all(by.css('.single-item'));
+  }
+
+  getOpenDialogButton() {
+    return element(by.css('.add-item'));
+  }
+
+  getModalDialog() {
+    return element(by.css('.dialog-modal'));
+  }
+
+  getCreateActionItemButton() {
+    return element(by.css('.create'));
+  }
+
+  getCloseDialogButton() {
+    return element(by.css('.close'));
+  }
+
+  getMatSelect() {
+    return element(by.tagName('mat-select'));
+  }
+
+  getMatOption() {
+    return element(by.cssContainingText('mat-option', 'CASD Wilson & Lamberton Middle Schools'));
+  }
+
+  getItemNameInput() {
+    return element(by.css('.name-input'));
+  }
+
+  getProjectInput() {
+    return element(by.css('.project-input'));
+  }
+
+  getDateInput() {
+    return element(by.css('.date-input'));
+  }
+
+  passActionItemName() {
```

Comment:
Please provide a unique name for the action item name. Currently, it is hard-coded - it is fine, but imagine situation when there is also `close` step in our e2e tests - (create action item, check if it's displayed in the table, then close it), and few people run these tests - it might be possible that there will be "test collision" - to avoid that please make action item unique per test execution - we need to generate some random unique value (guid or something) and use it as the name for the action item ( for example "Test Action Item - 1231345-65743-234").